src/App.jsx

```diff
@@ -14,57 +15,63 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const toggleGood = good => {
+    setSelectedGood(selectedGood === good ? '' : good);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  const clearSelection = () => {
+    setSelectedGood('');
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      {!selectedGood.length ? (
+        <h1 className="title is-flex is-align-items-center">
+          No goods selected
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood} is selected
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={clearSelection}
+          />
+        </h1>
+      )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              key={good}
+              className={
+                selectedGood === good && 'has-background-success-light'
+              }
             >
-              -
-            </button>
-          </td>
+              <td>
+                <button
+                  onClick={() => toggleGood(good)}
+                  data-cy={selectedGood !== good ? 'AddButton' : 'RemoveButton'}
+                  type="button"
+                  className={`button ${selectedGood === good && ' is-info'}`}
```

Comment:
Similar to the previous comment, the conditional class assignment `selectedGood === good && ' is-info'` will result in a `false` value being included in the `className` string when the condition is not met. Use a ternary operator to handle this: `selectedGood === good ? ' is-info' : ''`.